### PR TITLE
build: add unit test variable and variable expr

### PR DESCRIPTION
**Pull request by summer-ji-eng**

### CODE REVIEW

File: src/main/java/com/google/api/generator/engine/ast/Variable.java [hunk 1]
```diff
@@ -47,6 +47,7 @@ public Variable build() {
 
       Variable variable = autoBuild();
       Preconditions.checkState(!variable.type().typeKind().equals(TypeNode.TypeKind.VOID));
+      Preconditions.checkState(!variable.type().equals(TypeNode.NULL));
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/engine/ast/Variable.java:50]: Consider adding an error message, e,g. `checkState(condition, "Variable type cannot be null")`. Not part of this PR, but the VOID one above should have one as well.

Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/engine/ast/Variable.java:50]: Let me add in message in this PR.
